File: .claude/skills/agent-skill-kit-commons/packages/mjs/src/audit/printer.mjs
/**
 * Format the union of audit findings as either compact JSON or human text.
 * Findings are sorted into HIGH → MEDIUM → LOW buckets; within a bucket,
 * iteration order is preserved.
 */
export function formatAudit(findings, format = "human") {
  const counts = { HIGH: 0, MEDIUM: 0, LOW: 0 };
  for (const f of findings) {
    if (counts[f.severity] !== undefined) counts[f.severity]++;
  }
  if (format === "json") {
    return JSON.stringify(sortKeys({ summary: counts, findings })) + "\n";
  }
  let out = "";
  for (const sev of ["HIGH", "MEDIUM", "LOW"]) {
    for (const f of findings) {
      if (f.severity !== sev) continue;
      out += `[${sev}] ${f.code} ${f.where}: ${f.message}\n`;
    }
  }
  out += `\nsummary: ${counts.HIGH} HIGH, ${counts.MEDIUM} MEDIUM, ${counts.LOW} LOW\n`;
  return out;
}

function sortKeys(v) {
  if (Array.isArray(v)) return v.map(sortKeys);
  if (v && typeof v === "object") {
    const o = {};
    for (const k of Object.keys(v).sort()) o[k] = sortKeys(v[k]);
    return o;
  }
  return v;
}

File: .claude/skills/agent-skill-kit-commons/packages/py/src/skills_ref/audit/printer.py
from __future__ import annotations

import json


def format_audit(findings, fmt: str = "human") -> str:
    counts = {"HIGH": 0, "MEDIUM": 0, "LOW": 0}
    for f in findings:
        if f.get("severity") in counts:
            counts[f["severity"]] += 1
    if fmt == "json":
        out = {"summary": counts, "findings": findings}
        return json.dumps(_sort(out), sort_keys=True, separators=(",", ":")) + "\n"
    parts = []
    for sev in ("HIGH", "MEDIUM", "LOW"):
        for f in findings:
            if f.get("severity") != sev:
                continue
            parts.append(f"[{sev}] {f['code']} {f['where']}: {f['message']}\n")
    parts.append(
        f"\nsummary: {counts['HIGH']} HIGH, {counts['MEDIUM']} MEDIUM, {counts['LOW']} LOW\n"
    )
    return "".join(parts)


def _sort(v):
    if isinstance(v, list):
        return [_sort(x) for x in v]
    if isinstance(v, dict):
        return {k: _sort(v[k]) for k in sorted(v.keys())}
    return v
